src/services/testNumber.ts

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+
+import { InvokeCommand, LambdaClient } from '@aws-sdk/client-lambda';
+import logger from '../util/logger';
+import { dynamoDBClientConfig } from '../config';
+
+const lambdaClient = new LambdaClient({ region: dynamoDBClientConfig.region });
+
+export const generateSystemNumber = async (): Promise<string> => {
+  try {
+    if (process.env.AWS_SAM_LOCAL) {
+      return '123';
+    }
+    const input = {
+      path: '/system-number/',
+      httpMethod: 'POST',
+      resource: '/system-number/',
+    };
+
+    const command = new InvokeCommand({
+      FunctionName: process.env.TEST_NUMBER_LAMBDA_NAME,
+      InvocationType: 'RequestResponse',
+      Payload: JSON.stringify(input),
+    });
+
+    const response = await lambdaClient.send(command);
+    const bufferResponse = Buffer.from(response.Payload!).toString('utf-8');
+    const bufferBody = JSON.parse(bufferResponse).body;
+    return JSON.parse(bufferBody).systemNumber;
+  } catch (e) {
+    logger.error(`Error in generate system number ${JSON.stringify(e)}`);
+    throw new Error('lambda client failed getting data');
+  }
+};
```

Comment:
You should be able to write a generic function that just takes an argument of which number to generate, seeing as the logic is the same in all functions

```suggestion
export enum NumberTypes {
  ZNumber = "z-number",
  TrailerId = "trailerId",
  TNumber = "t-number",
  SystemNumber = "systemNumber"
}

// I wish this was not needed but test number service doesn't return a consistent object 
const apiResponseMap: Record<NumberTypes, string> {
  z-number: "ZNumber",
  trailerId: "trailerId",
  t-number: "TNumber",
  systemNumber: "systemNumber"
}


export const generateNewNumber = async (numberType: NumberTypes): Promise<string> => {
  try {
    if (process.env.AWS_SAM_LOCAL) {
      return '123';
    }
    const input = {
      path: `/${numberType}/`,
      httpMethod: 'POST',
      resource: '/`${numberType}/`,
    };
    const command = new InvokeCommand({
      FunctionName: process.env.TEST_NUMBER_LAMBDA_NAME,
      InvocationType: 'RequestResponse',
      Payload: JSON.stringify(input),
    });
    const response = await lambdaClient.send(command);
    const bufferResponse = Buffer.from(response.Payload!).toString('utf-8');
    const bufferBody = JSON.parse(bufferResponse).body;
    const apiResponseValue = apiResponseMap[numberType]
    return JSON.parse(bufferBody)[apiResponseValue];
  } catch (e) {
    logger.error(`Error in generate ${numberType} ${JSON.stringify(e)}`);
    throw new Error('lambda client failed getting data');
  }
};
```

And yes it would be nice if test number service return a consistent object with just a key of number for instance.
I mean it would actually be really nice if test number service wasn't so specific in its endpoints as well.